test/api.test.js

```diff
@@ -0,0 +1,111 @@
+import { test } from './helpers/context.js'
+
+import git from 'git-rev-sync'
+import pWaitFor from 'p-wait-for'
+import delay from 'delay'
+import { toString } from 'uint8arrays/to-string'
+import { Dealer } from '@web3-storage/filecoin-client'
+import { randomAggregate } from '@web3-storage/filecoin-api/test'
+
+import { getDealerClientConfig } from './helpers/dealer-client.js'
+import { getTableItem } from './helpers/table.js'
+import { getBucketItem } from './helpers/bucket.js'
+import {
+  getApiEndpoint,
+  getStage,
+  getDealStoreDynamoDb,
+  getOfferStoreBucketInfo
+} from './helpers/deployment.js'
+
+test.before(t => {
+  t.context = {
+    apiEndpoint: getApiEndpoint(),
+    dealStoreDynamo: getDealStoreDynamoDb(),
+    offerStoreBucket: getOfferStoreBucketInfo()
+  }
+})
+
+test('GET /version', async t => {
+  const stage = getStage()
+  const response = await fetch(`${t.context.apiEndpoint}/version`)
+  t.is(response.status, 200)
+
+  const body = await response.json()
+  t.is(body.env, stage)
+  t.is(body.commit, git.long('.'))
+})
+
+test('POST /', async t => {
+  const {
+    invocationConfig,
+    connection
+  } = await getDealerClientConfig(new URL(t.context.apiEndpoint))
+
+  // Create a random aggregate
+  const label = 'label'
+  const { pieces, aggregate } = await randomAggregate(100, 128)
+  const offer = pieces.map((p) => p.link)
+
+  const res = await Dealer.dealQueue(
+    invocationConfig,
+    aggregate.link.link(),
+    offer,
+    invocationConfig.with,
+    label,
+    { connection }
+  )
+
+  t.truthy(res.out.ok)
+  t.falsy(res.out.error)
+  t.truthy(aggregate.link.link().equals(res.out.ok?.aggregate))
+
+  // wait for deal-store entry to exist given it is propagated with a queue message
+  await delay(5_000)
+
+  console.log('try get deal entry...')
+  /** @type {import('../packages/core/src/data/deal.js').EncodedDeal | undefined} */
+  let dealEntry
+  await pWaitFor(async () => {
+    // @ts-expect-error does not automatically infer
+    dealEntry = await getTableItem(
+      t.context.dealStoreDynamo.client,
+      t.context.dealStoreDynamo.tableName,
+      { aggregate: res.out.ok?.aggregate?.toString() }
+    )
+
+    console.log('dealEntry found', Boolean(dealEntry))
+    return Boolean(dealEntry)
+  }, {
+    interval: 1_000,
+    timeout: 10_000
+  })
+  if (!dealEntry) {
+    throw new Error('deal store item was not found')
```

Comment:
I assume this is because of typescript? I want `t.assert(dealEntry, 'deal store item was not found')`

Reply:
yes, but that won't fix it. With `ava` it does not work that way to make types happy